sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -559,27 +560,39 @@ public HttpPipeline getHttpPipeline() {
      * @throws IllegalArgumentException If {@code blockSize} is less than 0 or greater than 100MB.
      * @throws UncheckedIOException If an I/O error occurs.
      */
-    public Mono<Void> downloadToFile(String filePath, BlobRange range, Integer blockSize,
-        ReliableDownloadOptions options, BlobAccessConditions accessConditions, boolean rangeGetContentMD5) {
-        return withContext(context -> downloadToFile(filePath, range, blockSize, options, accessConditions,
-            rangeGetContentMD5, context));
+    public Mono<Response<BlobProperties>> downloadToFileWithResponse(String filePath, BlobRange range,
+        Integer blockSize, ReliableDownloadOptions options, BlobAccessConditions accessConditions,
+        boolean rangeGetContentMD5) {
+        return withContext(context -> downloadToFileWithResponse(filePath, range, blockSize, options, accessConditions,
+                rangeGetContentMD5, context));
     }
 
-    Mono<Void> downloadToFile(String filePath, BlobRange range, Integer blockSize, ReliableDownloadOptions options,
-        BlobAccessConditions accessConditions, boolean rangeGetContentMD5, Context context) {
+    Mono<Response<BlobProperties>> downloadToFileWithResponse(String filePath, BlobRange range, Integer blockSize,
+                              ReliableDownloadOptions options, BlobAccessConditions accessConditions,
+                              boolean rangeGetContentMD5, Context context) {
         if (blockSize != null) {
             Utility.assertInBounds("blockSize", blockSize, 0, BLOB_MAX_DOWNLOAD_BLOCK_SIZE);
         }
 
         return Mono.using(() -> downloadToFileResourceSupplier(filePath),
-            channel -> Mono.justOrEmpty(range)
-                .switchIfEmpty(getFullBlobRange(accessConditions))
-                .flatMapMany(rg -> Flux.fromIterable(sliceBlobRange(rg, blockSize)))
-                .flatMap(chunk -> this.download(chunk, accessConditions, rangeGetContentMD5, context)
-                    .subscribeOn(Schedulers.elastic())
-                    .flatMap(dar -> FluxUtil.writeFile(dar.body(options), channel,
-                        chunk.getOffset() - (range == null ? 0 : range.getOffset()))))
-                .then(), this::downloadToFileCleanup);
+            channel -> blobDownloadInChunk(channel, range, blockSize, options, accessConditions, rangeGetContentMD5,
+                context),
+            this::downloadToFileCleanup);
+
+    }
+
+    private Mono<Response<BlobProperties>> blobDownloadInChunk(final AsynchronousFileChannel channel, BlobRange range,
+                              Integer blockSize, ReliableDownloadOptions options, BlobAccessConditions accessConditions,
+                              boolean rangeGetContentMD5, Context context) {
+        Mono<Response<BlobProperties>> asyncGetPropertiesResponse = getPropertiesWithResponse(accessConditions);
```

Comment:
The way this is being used may trigger two `getProperties` requests, can we verify this isn't happening.

Reply:
I don't think it triggered twice. The blob range used the response directly.

Reply:
Looking at this again the method doesn't get triggered to run until `.then` is called with it. `Monos` and `Fluxes` are deferred until a subscription is made.

https://projectreactor.io/docs/core/release/reference/#_operators

Reply:
From my experience, it will be evaluated twice.

Reply:
I have made a change on the stream calling. Could you take a look at the most recent code?

Reply:
Talked offline. Keep the most recent changes.